anndata/_core/anndata.py

```diff
@@ -1685,6 +1704,14 @@ def concatenate(
         ...     dict(obs_names=['s1', 's2'], anno2=['d3', 'd4']),
         ...     dict(var_names=['d', 'c', 'b'], annoA=[0, 2, 3], annoB=[0, 1, 2]),
         ... )
+
+        They can now be concatenated. Since this method is deprecated, it will raise a warning:
+
+        >>> warnings.filterwarnings(
+        ...     'ignore',
+        ...     r'The AnnData\\.concatenate method is deprecated',
+        ...     FutureWarning,
+        ... )
```

Comment:
I don't think we should show people how to turn off the warnings in documentation for deprecated methods. It suggests that they should turn off the warnings instead of just changing their usage.

Reply:
what’s your suggestion?

Reply:
For now, I'd suggest

```suggestion
```

But I've been eyeing pytest-doctestplus for a bit. `# doctest: +SHOW_WARNINGS` or `# doctest: +IGNORE_WARNINGS` could be used here in future.

Reply:
not possible, that’ll make that doctest fail in the --pre job

Reply:
I would also suggest not turning that on this PR.

Reply:
I invested quite some work into making it possible to do that in a way you’re happy with. Please don‘t go back on this now.

Reply:
As an alternative you can add this warning to the ignore.

I just want to merge the parts of this PR that are ready to go.

I don't want to add more into this PR because it touches a lot of orthogonal things. This makes it time consuming to review since I need to figure out which code change is related to which warning.

Reply:
> As an alternative you can add this specific warning to the ignore.

Which one? Not inline, you just said that. Not globally, we agreed that we want to be specific.

- Maybe for all doctests?
- Or by adding a `doctest_mark` or `doctest_filterwarnings` like in https://github.com/scverse/scanpy/pull/2605?

Reply:
done!